File: API/Entities/Movie.cs
using System.Text.Json.Serialization;

namespace API.Entities
{
  public class Movie
  {
    public int Id { get; set; }

    public string Title { get; set; }
    public string Genre { get; set; }
    public DateTime Premiere { get; set; }
    public string Description { get; set; }
    public string ImgUrl { get; set; }
    public int Duration { get; set; }
   public double AvgRating {get;set;}
    public List<Comment> Comments { get; set; } = new List<Comment>();
   
    public List<Rating> Ratings { get; set; } = new List<Rating>();
   
    public List<UserMovies> LikedByUsers { get; set; } = new List<UserMovies>();
  }
}

File: client/src/app/_models/Movie.ts
import { Comment } from "./Comment";
import { Like } from "./Like";
import { Rating } from "./Rating";

export interface Movie {
  id?: number;
  title: string;
  genre: string;
  rank?:number;
  premiere: Date;
  description: string;
  imgUrl: string;
  duration: number;
  avgRating:number;
  comments: Comment[];
  ratings: Rating[];
  likedByUsers:Like[]
}
